components/style/values/specified/outline.rs

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Specified values for outline properties
+
+use cssparser::Parser;
+use parser::{Parse, ParserContext};
+use selectors::parser::SelectorParseErrorKind;
+use style_traits::ParseError;
+use values::{Auto, Either};
+use values::specified::BorderStyle;
+
+/// <https://drafts.csswg.org/css-ui/#propdef-outline-style>
+pub type OutlineStyle = Either<Auto, BorderStyle>;
+
+impl OutlineStyle {
+    #[inline]
+    /// Get default value as None
+    pub fn second() -> OutlineStyle {
+        Either::Second(BorderStyle::None)
+    }
+
+    #[inline]
+    /// Get value for None or Hidden
+    pub fn none_or_hidden(&self) -> bool {
+        match *self {
+            Either::First(ref _auto) => false,
+            Either::Second(ref border_style) => border_style.none_or_hidden()
+        }
+    }
+
+    /// Parse outline-style
+    pub fn parse<'i, 't>(
```

Comment:
Hmm, so this is weird, how are we parsing `auto` now?

Before that, we used `Parse`, and then reject `BorderStyle::Hidden`, now that's not happening.

Let's change `OutlineStyle` to not be `Either`, but just be:

```rust
#[derive(ToCss, ToComputedValue, ...)]
pub enum OutlineStyle {
    Auto,
    Other(BorderStyle),
}
```

And parse `Auto` manually.

Reply:
Thanks!
I think using `enum` is more reasonable for this attribute.
I'm going to try it.